lib/server/ServerRequest.js

```diff
@@ -16,9 +18,16 @@ var ServerRequest = {
             environmentConfig: environmentConfig || {}
         };
     }
-
 };
 
+function getRawQuery(originalUrl) {
+    var index = originalUrl.indexOf('?');
+    if(originalUrl && index != -1) {
```

Comment:
the check for originalUrl is too late, if originalUrl is `null` or `undefined` line 24 would error out.

I propose

```js
function getRawQuery(originalUrl) {
    if (typeof originalUrl !== "string") {
       return null;
    }
    var indexOfQuestionMark = originalUrl.indexOf('?');
    if (indexOfQuestionMark === -1) {
        return null;
    }

    return originalUrl.substring(indexOfQuestionMark + 1);
}
```